File: Exercism/javascript/allergies/allergies.js
'use strict'

const ALLERGIES = [
  'eggs',
  'peanuts',
  'shellfish',
  'strawberries',
  'tomatoes',
  'chocolate',
  'pollen',
  'cats'
]

export class Allergies {
  constructor (score) {
    if (!Number.isInteger(score)) {
      throw new Error('Score is not an integer')
    }

    const gen = score
      .toString(2)
      .split('')
      .reverse()
      .map(Number)

    this.allergens = ALLERGIES
      .filter((e, i) => !!gen[i])
  }

  list () {
    return this.allergens
  }

  allergicTo (allergy) {
    return this.allergens.includes(allergy)
  }
}

File: Exercism/python/allergies/allergies.py
class Allergies(object):
    """Given a person's allergy score, determine whether or not
    they're allergic to a given item, and their full list of allergies."""
    ALLERGIES = [
            'eggs',
            'peanuts',
            'shellfish',
            'strawberries',
            'tomatoes',
            'chocolate',
            'pollen',
            'cats'
        ]

    def __init__(self, score):
        self.score = score

    def allergic_to(self, item: str) -> bool:
        """Whether or not the person is allergic to"""
        return bool(self.score & (1 << self.ALLERGIES.index(item)))

    @property
    def lst(self) -> list:
        """The full list of allergies"""
        return [a for a in self.ALLERGIES if self.allergic_to(a)]
